<!-- -*- mode: XML -*- -->
<launch>
	<arg name="time"/>
    <node name="dynamixel_manager_motors" pkg="dynamixel_controllers" type="controller_manager.py" required="true" output="screen">
        <rosparam>
            namespace: tflex_motors
            serial_ports:
                tflex_tilt_port:
                    port_name: "/dev/ttyUSB0"
                    baud_rate: 2000000
                    min_motor_id: 1
                    max_motor_id: 2
                    update_rate: 1000
        </rosparam>
    </node >
    <!-- Start tilt joint controller -->
		<rosparam file="$(find t_flex)/yaml/motors_parameters.yaml" command="load"/>
    <node name="tilt_controller_spawner_frontal" pkg="dynamixel_controllers" type="controller_spawner.py"
          args="--manager=tflex_motors
                --port tflex_tilt_port
                tilt1_controller"
                output="screen"/>
    <node name="tilt_controller_spawner_posterior" pkg="dynamixel_controllers" type="controller_spawner.py"
          args="--manager=tflex_motors
                --port tflex_tilt_port
                tilt2_controller"
                output="screen"/>
		<rosparam file="$(find t_flex)/yaml/online_detection.yaml" />
	<node pkg="t_flex" name="imu_data_acquisition" type="imu_data_acquisition.py" output ='screen'/>
	<node pkg="t_flex" name="real_time_gait_phase_det" type="real_time_gait_phase_det.py" output="screen"/>
	<node pkg="t_flex" name="gait_assistance" type="gait_assistance.py" output='screen' args="-t $(arg time)"/>
</launch>
